### DISCUSSION TITLE
`Air` object has no attribute `get_comp_vals`

### DISCUSSION THREAD

**Discussion by JorgeESantos** [PMEAL/OpenPNM#2714] [Q&A] [answered]
Greetings.

I'm trying to set up an Air phase in order to perform Fickian Diffusion simulation, to calculate Network tortuosity. The code I'm trying to run is:

```python
air = pnm.phase.Air(network=Net, name='air')
air['pore. Temperature'] = 293.15
air['throat. Temperature'] = 293.15 #24ºC = 297.15K
# set T = 20ºC
air.add_model(propname='throat.diffusive_size_factors',
              model=pnm.models.geometry.diffusive_size_factors.spheres_and_cylinders)
# air.add_model(propname='pore.diffusivity',
#               model=pnm.models.phase.diffusivity.gas_mixture_ce(phase=air, MWs=air.params['molecular_weight']))
# air.add_model(propname='throat.diffusivity',
#               model=pnm.models.phase.diffusivity.gas_mixture_ce(phase=air, MWs=air.params['molecular_weight']))
# tryed this way and nothing

air.add_model(propname='pore.diffusivity',
              model=pnm.models.phase.diffusivity.gas_mixture_ce(phase=air))
air.add_model(propname='throat.diffusivity',
              model=pnm.models.phase.diffusivity.gas_mixture_ce(phase=air))
air.add_model(propname='throat.diffusive_conductance',
              model=pnm.models.physics.diffusive_conductance.generic_diffusive)
```
The error raised is the following:
![image](https://user-images.githubusercontent.com/114251367/227233881-ba6bc3dd-6cfa-4acc-b317-6402884c3c83.png)

It seems I'm unable to define diffusity and that "gas_misture_ce" has not been able to fetch or reach air.params of Molecular Weight? With gas_mixture_fesg(), the result is the same.

Whats going on? Am I missing something?
@ma-sadeghi 

Regards

**Comment by ma-sadeghi** [comment, marked as answer]
Hi. There are a couple of issues with your code:

- For starters, when you add a model to a phase, you just need to pass a "handle" to the model, and not call it. In your case, you just need to pass `gas_mixture_ce` instead of `gas_mixture_ce(...)`.
- `Air` is already a mixture so I assume it should already have a model for diffusivity, so I guess you don't need to add those models yourself.
- When setting temperature, use lowercase, i.e., `temperature` instead of `Temperature`.
- Size factor models are to be added to your network, and not to the phase object.

Have you looked at [this](https://openpnm.org/examples/applications/effective_diffusivity_and_tortuosity.html) example? It explains how to compute tortuosity of a network.

**Comment by JorgeESantos** [comment]
Hey there. Thanks for the quick answer. Its solved. I added the diffusive size factors to the Network and then, I deleted diffusivity models and it does the trick. The temperature was a typo when I wrote it on Github. Now the code is:

```python
air = pnm.phase.Air(network=Net, name='air')
air.add_model_collection(pnm.models.collections.physics.basic)
air['pore.temperature'] = 293.15
air['throat.temperature'] = 293.15 #24ºC = 297.15K
# set T = 20ºC
air.add_model(propname='throat.diffusive_conductance',
              model=pnm.models.physics.diffusive_conductance.generic_diffusive)
print('Phase and physics set')
```

Thanks for the help. Should I delete the thread?

**Reply by ma-sadeghi** [reply]
No worries, I just moved it to the Discussions tab.
